File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> leftProd(n,0);
        vector<int> rightProd(n,0);
         vector<int> ans(n,0);
        
        //find left prod
        leftProd[0] = 1;
        for(int i=1;i<n;i++){
            leftProd[i] = leftProd[i-1] * nums[i-1];
        }

        //find right prod
        rightProd[n-1] = 1;
        for(int i=n-2;i>=0;i--){
            rightProd[i] = rightProd[i+1] * nums[i+1];
        }

        for(int i=0;i<n;i++){
            ans[i] = leftProd[i] * rightProd[i];
        }

        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1 for i in range(0,len(nums))]
        right = [1 for i in range(0,len(nums))]
        ans = []
        # left mul
        for i in range(1,len(nums)):
            left[i] = left[i-1] * nums[i-1]
        
        # right mul
        for i in range(len(nums)-2,-1,-1):
            right[i] = right[i+1] * nums[i+1]
        
        for i in range(0,len(nums)):
            ans.append(left[i] * right[i])
        
        return ans
        